include/deal.II/distributed/tria.h

```diff
@@ -818,6 +818,55 @@ namespace parallel
       add_periodicity
       (const std::vector<GridTools::PeriodicFacePair<cell_iterator> > &);
 
+      template<typename T>
+      struct sum
+      {
+        typedef T result_type;
+
+        template<typename InputIterator>
+        T operator()(InputIterator first, InputIterator last) const
+        {
+          // If there are no slots to call, just return the
+          // default-constructed value
+          if (first == last)
+            return T();
+
+          T sum = *first++;
+          while (first != last)
+            {
+              sum += *first++;
+            }
+
+          return sum;
+        }
+      };
+
+      /**
+       * A structure that has boost::signal objects for a number of actions that a
+       * triangulation can do to itself. See the general documentation of the
+       * Triangulation class for more information and for documentation of the
+       * semantics of the member functions.
+       *
+       * For documentation on signals, see
+       * http://www.boost.org/doc/libs/release/libs/signals2 .
+       */
+      struct Signals
+      {
+        boost::signals2::signal<void ()> create;
+        boost::signals2::signal<void ()> pre_refinement;
+        boost::signals2::signal<void ()> post_refinement;
+        boost::signals2::signal<void (const Triangulation<dim, spacedim> &original_tria)> copy;
+        boost::signals2::signal<void ()> clear;
+        boost::signals2::signal<void ()> any_change;
+        boost::signals2::signal<unsigned int (const cell_iterator &,
+                                              const CellStatus),Triangulation<dim,spacedim>::sum<unsigned int> > cell_weight;
```

Comment:
Put the accumulator template argument onto its own line to make it easier to read.